sdcm/nemesis.py

```diff
@@ -1434,8 +1434,21 @@ def disrupt_kill_scylla(self):
     def disrupt_no_corrupt_repair(self):
         # prepare test tables and fill test data
         for i in range(10):
+            ks_name = f"drop_table_during_repair_ks_{i}"
+            test_keyspaces = self.cluster.get_test_keyspaces()
```

Comment:
I would rather have this call outside of the loop.

In cases we have lots of keyspaces it could be quite slow and problematic to call it so many times

Reply:
Also did you considered add this option of precreate the keyspaces into `_prepare_test_table`

Reply:
And also this change only create keyspace, but creation of the
 table is also schema change we might want also to have  the table created in advance

Reply:
@fruch  Initially, I was thinking about changing `_prepare_test_table`, but it is used by several nemeses, not only by `disrupt_no_corrupt_repair`, so after a short discussion with @roydahan I made the change only in `disrupt_no_corrupt_repair`.
If we also need to create the tables in advance for `disrupt_no_corrupt_repair` then I guess I have no better option than change `_prepare_test_table`. Following your advice, I could introduce a new flag argument for `_prepare_test_table` and create objects in advance there only if it is `True`.

Reply:
that's what I was point to, to extend the API of that call.

Reply:
Removed ks and table creation from the nemesis and added a new argument for `_prepare_test_table` method - `pre_create_objects` which is `True` only while calling the method from this nemesis (and `False` by default).